crates/iota-sdk-ffi/src/crypto/ed25519.rs

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use iota_types::SignatureScheme;
+
+use crate::{
+    error::{Result, SdkFfiError},
+    types::{
+        crypto::{Ed25519PublicKey, Ed25519Signature},
+        signature::{SimpleSignature, UserSignature},
+    },
+};
+
+#[derive(derive_more::From, uniffi::Object)]
+pub struct Ed25519PrivateKey(iota_crypto::ed25519::Ed25519PrivateKey);
+
+#[uniffi::export]
+impl Ed25519PrivateKey {
+    #[uniffi::constructor]
+    pub fn new(bytes: Vec<u8>) -> Self {
+        Self(iota_crypto::ed25519::Ed25519PrivateKey::new(
+            bytes.try_into().unwrap(),
+        ))
+    }
+
+    pub fn scheme(&self) -> SignatureScheme {
+        self.0.scheme()
+    }
+
+    pub fn verifying_key(&self) -> Ed25519VerifyingKey {
+        self.0.verifying_key().into()
+    }
+
+    pub fn public_key(&self) -> Ed25519PublicKey {
+        self.0.public_key().into()
+    }
+
+    #[uniffi::constructor]
+    pub fn generate() -> Self {
+        use rand::rngs::OsRng;
+        Self(iota_crypto::ed25519::Ed25519PrivateKey::generate(OsRng))
+    }
+
+    /// Deserialize PKCS#8 private key from ASN.1 DER-encoded data (binary
+    /// format).
+    #[uniffi::constructor]
+    pub fn from_der(bytes: &[u8]) -> Result<Self> {
+        iota_crypto::ed25519::Ed25519PrivateKey::from_der(bytes)
+            .map(Self)
+            .map_err(SdkFfiError::custom)
```

Comment:
I prefer this syntax. Thoughts?
```suggestion
        Ok(iota_crypto::ed25519::Ed25519PrivateKey::from_der(bytes)?.into())
```

Reply:
https://github.com/iotaledger/iota-rust-sdk/pull/127/commits/33564549bb9c81abd338a9e16dcc9f44c2ed3c5d